micro-rdk-server/Cargo.toml

```diff
@@ -55,3 +55,8 @@ esp_idf_sdkconfig_defaults = [
   "micro-rdk-server/sdkconfig.defaults",
   "micro-rdk-server/sdkconfig.defaults.esp32"
 ]
+
+[[package.metadata.esp-idf-sys.extra_components]]
```

Comment:
I'm not sure. According to the [`esp-idf-sys` build options](https://github.com/esp-rs/esp-idf-sys/blob/master/BUILD-OPTIONS.md), this feature needs to be in the Cargo.toml of the workspace or `ESP_IDF_SYS_ROOT_CRATE`, [link](https://github.com/esp-rs/esp-idf-sys/blob/master/BUILD-OPTIONS.md#esp-idf-configuration)

Also, we should note that this modifies the build of `esp-idf-sys` that gets linked to by everything in our workspace.
I don't know how this would affect a user who is using `cargo-generate` to build a project.

Reply:
We have two options
- user adds block via camera option on cargo generate (recently added)
- we add this block into `micro-rdk` crate but will be included in `esp-idf-sys` even when the `camera` feature is not enabled. Whether this is optimized out by the compiler or not is unclear.

Reply:
to clarify, should I remove this or comment it out and add a note to enable when developing builtin-camera?

I'm not sure there's a way to control this block from here with a feature-gate

Reply:
leaving in as per offline conversation. builds in docker